src/main/ts/utils/ObjUtils.ts

```diff
@@ -5,7 +5,7 @@ type Option<A> = O.Option<A>;
 export const hasKey = <T>(o: T, k: keyof T): boolean =>
   Object.prototype.hasOwnProperty.call(o, k);
 
-export const lookup = <T>(o: T, k: keyof T): Option<T[keyof T]> =>
+export const lookup = <K extends keyof T, T>(o: T, k: K): Option<T[K]> =>
```

Comment:
Didn't really mean to commit this, however it does ensure the types from lookup are more correct as previous it was doing a union of all possible values in `T`.